PreFreeSurfer/PreFreeSurferPipeline.sh

```diff
@@ -452,253 +531,327 @@ log_Msg "POSIXLY_CORRECT="${POSIXLY_CORRECT}
+    convertwarp --relout --rel --ref=${T1wTemplate} --premat=${T2wFolder}/xfms/acpc.mat --warp1=${T1wFolder}/xfms/${T2wImage}_reg_dc --out=${T1wFolder}/xfms/${OutputOrigT2wToT1w}
+
+    OutputT2wImage=${T1wFolder}/${T2wImage}_acpc_dc
+    applywarp --rel --interp=spline -i ${T2wFolder}/${T2wImage} -r ${T1wTemplate} -w ${T1wFolder}/xfms/${OutputOrigT2wToT1w} -o ${OutputT2wImage}
+    fslmaths ${OutputT2wImage} -abs ${OutputT2wImage} -odt float  # Use -abs (rather than '-thr 0') to avoid introducing zeros
+    fslmaths ${OutputT2wImage} -div ${T1wFolder}/BiasField_acpc_dc ${OutputT2wImage}_restore
+    fslmaths ${OutputT2wImage}_restore -mas ${T1wFolder}/${T1wImage}_acpc_dc_brain ${OutputT2wImage}_restore_brain
 
-log_Msg "Creating one-step resampled version of {T1w,T2w}_acpc_dc outputs"
+    # Remove the file (warpfield) that serves as a proxy in FreeSurferPipeline for whether PostFreeSurfer has been run
+    # i.e., whether the T1w/T2w_acpc_dc* volumes reflect the PreFreeSurferPipeline versions (above)
+    # or the PostFreeSurferPipeline versions
+    OutputOrigT2wToT1wPostFS=OrigT2w2T1w  #Needs to match name used in both FreeSurferPipeline and PostFreeSurferPipeline
+    imrm ${OutputOrigT2wToT1wPostFS}
+  fi
 
-# T1w
-OutputOrigT1wToT1w=OrigT1w2T1w_PreFS  # Name for one-step resample warpfield
-convertwarp --relout --rel --ref=${T1wTemplate} --premat=${T1wFolder}/xfms/acpc.mat --warp1=${T1wFolder}/xfms/${T1wImage}_dc --out=${T1wFolder}/xfms/${OutputOrigT1wToT1w}
 
-OutputT1wImage=${T1wFolder}/${T1wImage}_acpc_dc
-applywarp --rel --interp=spline -i ${T1wFolder}/${T1wImage} -r ${T1wTemplate} -w ${T1wFolder}/xfms/${OutputOrigT1wToT1w} -o ${OutputT1wImage}
-fslmaths ${OutputT1wImage} -abs ${OutputT1wImage} -odt float  # Use -abs (rather than '-thr 0') to avoid introducing zeros
-fslmaths ${OutputT1wImage} -div ${T1wFolder}/BiasField_acpc_dc ${OutputT1wImage}_restore
-fslmaths ${OutputT1wImage}_restore -mas ${T1wFolder}/${T1wImage}_acpc_dc_brain ${OutputT1wImage}_restore_brain
+# -- Are we using a custom mask?
 
-#T2w
-OutputOrigT2wToT1w=OrigT2w2T1w_PreFS  # Name for one-step resample warpfield
-convertwarp --relout --rel --ref=${T1wTemplate} --premat=${T2wFolder}/xfms/acpc.mat --warp1=${T1wFolder}/xfms/${T2wImage}_reg_dc --out=${T1wFolder}/xfms/${OutputOrigT2wToT1w}
+elif [ "$CustomBrain" = "MASK" ] ; then
 
-OutputT2wImage=${T1wFolder}/${T2wImage}_acpc_dc
-applywarp --rel --interp=spline -i ${T2wFolder}/${T2wImage} -r ${T1wTemplate} -w ${T1wFolder}/xfms/${OutputOrigT2wToT1w} -o ${OutputT2wImage}
-fslmaths ${OutputT2wImage} -abs ${OutputT2wImage} -odt float  # Use -abs (rather than '-thr 0') to avoid introducing zeros
-fslmaths ${OutputT2wImage} -div ${T1wFolder}/BiasField_acpc_dc ${OutputT2wImage}_restore
-fslmaths ${OutputT2wImage}_restore -mas ${T1wFolder}/${T1wImage}_acpc_dc_brain ${OutputT2wImage}_restore_brain
+  log_Msg "Skipping all the steps to Atlas registration, applying custom mask."
+  ceho "---> Applying custom mask"
```

Comment:
What is `ceho`?  And regardless, why not using `log_Msg`?

Reply:
`ceho` is a colored version of echo. We’ve had this discussion in the previous debugging scrip commit and reached a consensus that I’ll implement. The reason for not using `log_Msg` was that it is easier to see when you are reviewing the output in real time. Also I believe we introduced progress tracking in this manner before the “new” logging functionality was introduced and kept it for legibility reasons.

Reply:
To illustrate why ceho and the verbose echo statements in some places, please see the attached image of an output, which shows how easier it is to follow the progress in this manner.

![ceho_and_verbosity](https://user-images.githubusercontent.com/1247439/65815584-c2053580-e1f1-11e9-9d85-54ff12a52223.png)

Reply:
It seems very inconsistent to me to start using `echo` rather than `log_Msg`.  What is your proposal for when to use each?

Reply:
Woud this be acceptable:

We use `log_Msg` wherever it was used before. We introduce two system variables, one that controls coloring (as discussed previously in the debugging pull request) and one that turns on the debug mode. I then replace `echo` lines that we added with e.g. `decho` and they are only printed when the debug mode is turned on.

Reply:
We already have a `log_Debug_Msg()` function.  Can that be used?  Otherwise we now have two different functions that are related to providing extra verbosity in a "debug" mode.

Reply:
The difference is in the specific use and the display. The new `verbose_echo` is much more detailed and w/o any other information but the steps done. I'd prefer staying with the two options.